### DISCUSSION TITLE
Calculate JSON output size without serializing

### DISCUSSION THREAD

**Discussion by eqbalz** [open-source-parsers/jsoncpp#1463] [Q&A]
I have a requirement where I need to create a batch of JSON entries and flush them to disk if the next element would cause it to reach above a certain size threshold. The size() is not good enough as my arrays could have nested JSON values of varying types (string, int, boolean etc.). Is there any way to compute the physical size of the JSON string that would be produced once we serialize it without first having to create the string first? I am trying to save some copying and memory.

**Comment by BillyDonahue** [comment]
I don't think so. I mean you could try and make a reasonable guess.
Maybe the copying and memory aren't so bad.
